driver/ni/ai_channels.h

```diff
@@ -0,0 +1,1316 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+
+#pragma once
+
+#include <string>
+
+#include "daqmx.h"
+#include "nidaqmx_api.h"
+#include "nisyscfg.h"
+#include "client/cpp/telem/telem.h"
+#include "driver/config/config.h"
+#include "glog/logging.h"
+#include "nlohmann/json.hpp"
+
+namespace ni {
+typedef struct ExcitationConfig {
+    int32_t voltageExcitSource;
+    double voltageExcitVal;
+    double minValForExcitation; // optional
+    double maxValForExcitation; //optional
+    bool32 useExcitForScaling; //optional
+
+    static inline int32_t getExcitationSrc(std::string s){
+        if(s == "Internal") return DAQmx_Val_Internal;
+        if(s == "External") return DAQmx_Val_External;
+        if(s == "None") return DAQmx_Val_None;
+        return DAQmx_Val_None;
+    }
+    
+    ExcitationConfig(config::Parser &parser)
+        : voltageExcitSource(getExcitationSrc(parser.required<std::string>("voltage_excit_source"))),
+          voltageExcitVal(parser.required<double>("voltage_excit_val")),
+          minValForExcitation(parser.optional<double>("min_val_for_excitation", 0)),
+          maxValForExcitation(parser.optional<double>("max_val_for_excitation", 0)),
+          useExcitForScaling(parser.optional<bool32>("use_excit_for_scaling", 0)) {
+    }
+} ExcitationConfig;
+
+typedef struct PolynomialConfig {
+    float64 *forwardCoeffs;
+    uint32_t numForwardCoeffs;
+    float64 *reverseCoeffs;
+    uint32_t numReverseCoeffs;
+    int32_t electricalUnits;
+    int32_t physicalUnits;
+
+    PolynomialConfig() = default;
+
+    PolynomialConfig(config::Parser &parser)
+        : numForwardCoeffs(parser.required<uint32_t>("num_forward_coeffs")),
+          numReverseCoeffs(parser.required<uint32_t>("num_reverse_coeffs")),
+          electricalUnits(parser.required<int32_t>("electrical_units")),
+          physicalUnits(parser.required<int32_t>("physical_units")) {
+        if (!parser.ok()) return; // TODO: handle error
+
+        json j = parser.get_json();
+
+        forwardCoeffs = new double[numForwardCoeffs];
+        reverseCoeffs = new double[numReverseCoeffs];
+
+        //get forward coeffs (prescale -> scale)
+        if (j.contains("forward_coeffs")) {
```

Comment:
Is this necessary? Isn't there a way we can access the underlying array directly from the vector?